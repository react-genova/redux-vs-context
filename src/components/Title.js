import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Title = ({ title, className }) => {
    console.log('render Title');
    return <h1 className={className}>{title}</h1>;
};

Title.propTypes = {
    className: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
};

const TitleStyled = styled(Title)`
    text-align: center;
    margin: 10px 0 0 0;
    color: #afa;
`;

export default TitleStyled;
