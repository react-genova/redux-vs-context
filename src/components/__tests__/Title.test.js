import React from 'react';
import Title from '../Title';
import { render } from 'react-testing-library';
import 'jest-styled-components';

describe('Title component', () => {
    it('renders without crashing', () => {
        const { container } = render(
            <Title title="the title" className="theclassname" />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
