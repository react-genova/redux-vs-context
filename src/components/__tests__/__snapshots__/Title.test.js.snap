// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component renders without crashing 1`] = `
.c0 {
  text-align: center;
  margin: 10px 0 0 0;
  color: #afa;
}

<h1
  class="theclassname c0"
>
  the title
</h1>
`;
